src/renderer/dx/DxRenderer.cpp

```diff
@@ -1352,6 +1366,47 @@ try
 }
 CATCH_RETURN()
 
+// Routine description:
+// - Actual painting procedure
+// Arguments:
+// - <none>
+// Return Value:
+// - Any DirectX error, a memory error, etc.
+[[nodiscard]] HRESULT DxEngine::PaintFrame(IRenderData* pData) noexcept
+try
+{
+    if (pData == nullptr)
+    {
+        return S_OK;
+    }
+
+    // Prep Colors
+    RETURN_IF_FAILED(UpdateDrawingBrushes(pData->GetDefaultBrushColors(), pData, true));
+
+    // Prepare the engine with additional information before we start drawing.
+    RenderFrameInfo info;
+    info.cursorInfo = _GetCursorInfo(pData);
+    RETURN_IF_FAILED(PrepareRenderInfo(info));
+
+    // Paint Background
+    RETURN_IF_FAILED(PaintBackground());
+
+    // Paint Rows of Text
+    _LoopDirtyLines(pData, std::bind(&DxEngine::_PaintBufferLineHelper, this, std::placeholders::_1));

```

Comment:
Last I checked... `std::bind` is not cheap.

Reply:
I thought so, too. But the trace shows that it's OK.